### PR TITLE
Features/3 firebase auth

### PR DESCRIPTION
**Pull request by stanislav-ostr**
https://trello.com/c/pJcHFPge/3-firebase-authentication-setup

### CODE REVIEW

File: frontend/src/app/core/services/firebase-auth.service.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+export class FirebaseAuthService {
+    private tokenKeyName = 'user-token';
+
+    constructor(private afAuth: AngularFireAuth, private ngZone: NgZone) {
+        afAuth.authState.subscribe(async (user) => {
+            if (user) {
+                localStorage.setItem(this.tokenKeyName, await user.getIdToken());
+            } else {
+                localStorage.removeItem(this.tokenKeyName);
+            }
```
Comment (tatianahutii) [inline frontend/src/app/core/services/firebase-auth.service.ts:20]:
```suggestion
            user
                ? localStorage.setItem(this.tokenKeyName, await user.getIdToken())
                :  localStorage.removeItem(this.tokenKeyName);
```

Comment (stanislav-ostr) [reply frontend/src/app/core/services/firebase-auth.service.ts:20]: ### ESLint doesn't allow it

Screenshot 2023-08-22 165351
Screenshot 2023-08-22 165431

File: backend/LeetWars.Core/LeetWars.Core.WebAPI/Extentions/ServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -40,5 +42,22 @@ public static void AddLeetWarsCoreContext(this IServiceCollection services, ICon
                     connectionsString,
                     opt => opt.MigrationsAssembly(typeof(LeetWarsCoreContext).Assembly.GetName().Name)));
         }
+
+        public static void AddFirebaseAuthentication(this IServiceCollection services)
+        {
+            services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
+                .AddJwtBearer(options =>
+                {
+                    options.Authority = "https://securetoken.google.com/leetwars";
```
Comment (tatianahutii) [inline backend/LeetWars.Core/LeetWars.Core.WebAPI/Extentions/ServiceCollectionExtensions.cs:51]: can you move this to appsettings and use in a both places

File: frontend/angular.json [hunk 3]
```diff
@@ -1,126 +1,143 @@
...
-                },
-                "@schematics/angular:guard": {
-                    "skipTests": true
-                },
-                "@schematics/angular:pipe": {
-                    "skipTests": true
+  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
+  "cli": {
+    "analytics": false,
+    "schematicCollections": [
```
Comment (tatianahutii) [inline frontend/angular.json:5]: toooo many changes, can you some how add only your changes?

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: frontend/angular.json (hunk 3 above)
Comment (stanislav-ostr) [reply frontend/angular.json:5]: It was only one command

image

**Review by tatianahutii — APPROVED** [review]
LGTM

**Review by Den313 — APPROVED** [review]
LGTM

**Pull request merged**